### PR TITLE
Fixed regression where player type was not being reported with player initialization 

### PR DESCRIPTION
**Pull request by esthersweon**
**User-Facing Changes and Description**
- Fixed regression where player type was not being reported with player initialization 
- Also removed Foxglove Data Platform connection from the desktop app menu's list of connection options
- Also fixed formatting of ampersand in the desktop app's File menu ("Rosbridge (ROS 1 & 2)")

Screen Shot 2021-12-02 at 3 19 55 PM

### CODE REVIEW

File: desktop/main/StudioWindow.ts [hunk 1]
```diff
@@ -391,14 +391,22 @@ class StudioWindow {
   }
 
   addInputSource(name: string): void {
-    this._inputSources.add(name);
+    const unsupportedInputSourceNames = ["Foxglove Data Platform"];
```
Comment (jhurliman) [inline desktop/main/StudioWindow.ts:394]: Why is this unsupported? Comment?

File: desktop/main/StudioWindow.ts [hunk 2]
```diff
@@ -391,14 +391,22 @@ class StudioWindow {
...
+    if (unsupportedInputSourceNames.includes(name)) {
+      return;
+    }
+    const formattedName = name
+      .split("")
+      .map((char) => (char === "&" ? "&&" : char))
```
Comment (jhurliman) [inline desktop/main/StudioWindow.ts:400]: I can see this is doubling the number of ampersands, but I'm not sure why.

Comment (esthersweon) [reply desktop/main/StudioWindow.ts:400]: https://github.com/electron/electron/issues/25028, which led me to https://www.electronjs.org/docs/latest/api/menu#static-methods.

Will add comment.

File: desktop/main/StudioWindow.ts (hunk 1 above)
Comment (esthersweon) [reply desktop/main/StudioWindow.ts:394]: Adding the following:

```
 // A "Foxglove Data Platform" connection is triggered by opening a URL from console
 // Not currently a connection that can be started from inside Foxglove Studio
```

**Pull request merged**